File: evidence/generated-systems/angular/hr-on-angular/src/app/models/TrainingCourse.js

// Define collection and schema for TrainingCourse
export interface TrainingCourse {
    code:
	type : string
    title:
	type : string
    durationHours:
	type : String
    Prerequisites:
 	type : [{ type: Schema.Types.ObjectId, ref: 'TrainingCourse' }]
    Enrollments:
 	type : [{ type: Schema.Types.ObjectId, ref: 'TrainingEnrollment' }]
    JobProfiles:
 	type : [{ type: Schema.Types.ObjectId, ref: 'JobProfile' }]
    DeliveryMethod:
 	type : String
#
    collection: 'trainingCourses'
}

File: evidence/generated-systems/golang/hr-on-golang/hr-on-golang/internal/model/TrainingCourse.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// TrainingCourse Declaration
//==============================================================
type TrainingCourse struct {
    gorm.Model
     Code                                    string
    Title                                    string
    DurationHours                                                            string
     Prerequisites           []TrainingCourse `gorm:"foreignKey:PrerequisitesFromTrainingCourseId"`
     Enrollments           []TrainingEnrollment `gorm:"foreignKey:EnrollmentsFromTrainingCourseId"`
     JobProfiles           []JobProfile `gorm:"foreignKey:JobProfilesFromTrainingCourseId"`
    DeliveryMethod                      DeliveryMethod

// parent associations as their child

}

